src/main/main.ts

```diff
@@ -7,6 +7,48 @@ import log from 'electron-log';
 import * as yargs from 'yargs';
 import * as path from 'path';
 import * as fs from 'fs';
+import { randomBytes } from 'crypto';
+import { createServer } from 'net';
+
+import { appConfig } from './utils';
+
+async function getFreePort(startPort: number): Promise<number> {
+    return new Promise<number>((resolve) => {
+        let nextPort = startPort;
+
+        const getPort = () => {
+            let port = nextPort++;
+
+            const server = createServer((socket) => {
+                socket.write('Echo server\r\n');
+                socket.pipe(socket);
+            });
+            
+            server.on('error', function (e) {
+                getPort();
+            });
+            server.on('listening', function (e: any) {
+                server.close();
+                
+                resolve(port);
+            });
+        
+            server.listen(port, '127.0.0.1');
+        };
+
+        getPort();
+    });
+};
```

Comment:
The Node.js documentation says:

> If port is omitted or is 0, the operating system will assign an arbitrary unused port, which can be retrieved by using `server.address().port`

Could we delegate the responsibility of choosing the port to the operating system to avoid potential issues?